UPP Operations

AndroBetel opened the pull request:
Adds new shipmap - SSV Chapaev along with NAVAL INFANTRY Platoon "Red Dawn"
Buffs some UPP weapons to be on par with PVE USCM weapons
Gives new button to GMs that allows them to choose which type of platoon to deploy (only two are available rn - upp and default one)

FslashN commented:
I've looked through it and everything appears to be present. Definitely needs to be tested in a few games to make sure everything is working correctly, but otherwise I don't see any immediate problems. The GM button allows one to change what map/platoon is used when the game round has not started, which is fine.

My only real concern is that people will use their marine characters on this, which can be immersion breaking.

AndroBetel commented:
i consider this pr to be ready for testing!

FslashN commented:
> i consider this pr to be ready for testing!

Cool. It is currently test merged and should be available for GMs.

FslashN commented:
There was another round this morning, and this is some mechanical feedback I have. I'll edit this if I run into more stuff.

- ColMar food vendors and synth vendors still exist. Need to be renamed and description changed. UPP doesn't use WY either, if I am not mistaken, so the medical vendors could be changed too. Vendors could use new sprites overall, but that's secondary and not important.
- Mob faction should not be USCM, as we may want to spawn these together with marines at some point, and they should not share IFF. All access levels should also be unique from the Golden Arrow for similar reasons.
- On that note, because the OW consoles are set to USCM, marine raiders show up in there. OW tracking should be UPP only, so the UPP can't see marines on the ground/elsewhere and vice versa.
- Motion detectors should be the hacked variant that has UPP as the faction, not the default.
- Also still seeing some USCM items, like the M276 pattern medical storage rig.

morrowwolf commented on maps/chapaev.json:
```diff
@@ -0,0 +1,8 @@
+{
+    "map_name": "SSV Chapaev",
```
Yo, the setting for it being UPP needs to be from here rather than a setting that is ticked somewhere. We never want marines on this ship map, we never want UPP on the other ship map.

FslashN reviewed (changes requested):
There was another round this morning, and this is some mechanical and flavor feedback I have. I'll edit this if I run into more stuff. This is off [903b8cd](https://github.com/PvE-CMSS13/PvE-CMSS13/pull/108/commits/903b8cdcd9b1030eb1cffc780d3fe3d5266ca779) which was compiled in testmerge yesterday.

- ColMar food vendors and synth vendors still exist. Need to be renamed and description changed. UPP doesn't use WY either, if I am not mistaken, so the medical vendors could be changed too. Vendors could use new sprites overall, but that's secondary and not important.
- Mob faction should not be USCM, as we may want to spawn these together with marines at some point, and they should not share IFF. All access levels should also be unique from the Golden Arrow for similar reasons.
- On that note, because the OW consoles are set to USCM, marine raiders show up in there. OW tracking should be UPP only, so the UPP can't see marines on the ground/elsewhere and vice versa.
- Motion detectors should be the hacked variant that has UPP as the faction, not the default.
- Also still seeing some USCM items, like the M276 pattern medical storage rig.

AndroBetel commented:
most vendors are now customized to fit UPP background
mob faction changed to UPP, same with machinery across the ship and MDs
i changed the medical belt, but i'm not a spriter and for now we'll probably have to cope with lrp marine items...

AndroBetel commented:
morrow's review done!

AndroBetel commented:
updated sprites by amory !

FslashN commented on code/game/area/Sulaco.dm:
```diff
@@ -123,6 +123,18 @@
 	return TRUE
 
+/area/shuttle/ds_upp
+	name = "\improper Dropship Akademia Nauk"
```
I think this should be `name = "dropship Akademia Nauk"`

FslashN replied:
I think this should be `name = "dropship Akademia Nauk"`.

FslashN commented on code/game/jobs/job/marine/squads.dm:
```diff
@@ -129,6 +129,22 @@
+	name = SQUAD_UPP
+	equipment_color = "#8B0000"
+	chat_color = "#B22222"
+	access = list(ACCESS_MARINE_ALPHA)
```
This should be a unique access define. If they end up on the Golden Arrow or elsewhere with marine access, we don't want them to actually have it.

FslashN commented on code/game/jobs/job/marine/squads.dm:
```diff
@@ -129,6 +129,22 @@
+	chat_color = "#B22222"
+	access = list(ACCESS_MARINE_ALPHA)
+	radio_freq = UPP_FREQ
+	minimap_color = MINIMAP_SQUAD_ALPHA
```
Same here, unique minimap color; though if I'm not mistaken, don't we already have something UPP related for the map?

FslashN reviewed (changes requested):
A few minor things. Looks like you've addressed just about everything. Like the PKP changes with the added action, something else I was going to mention.

FslashN reviewed (dismissed):
Most of the things specified were addressed.

FslashN reviewed (dismissed):
Changes are fine. Need to test merge again to see that everything work as intended.

FslashN commented on code/modules/mob/living/carbon/human/human.dm:
```diff
@@ -1076,12 +1076,11 @@
 	set category = "IC"
 
-	if(faction != FACTION_MARINE && !(faction in FACTION_LIST_WY))
+	if(job in ROLES_AI)
```
This will prevent roles that are not Distress Signal (Lowpop) from accessing the manifest. So if you are playing a game mode that changes the roles, it will not work right.
`if(job in SSticker.mode.get_roles_list())`

FslashN commented on code/controllers/subsystem/minimap.dm:
```diff
@@ -907,7 +907,7 @@ SUBSYSTEM_DEF(minimaps)
 			old_map = get_tacmap_data_png(faction)
 
-			if(faction == FACTION_MARINE)
+			if(faction in FACTION_LIST_HUMANOID )
```
A lot of these checks should probably take into account if the user faction matches the faction of the CICmap for it to function properly.
`if(faction in FACTION_LIST_HUMANOID && user.faction == faction)`

There is also a function later on:
`notify_ghosts(header = "Tactical Map", message = "The USCM tactical map has been updated.", ghost_sound = "sound/effects/sos-morse-code.ogg", notify_volume = 80, action = NOTIFY_USCM_TACMAP, enter_link = "uscm_tacmap=1", enter_text = "View", source = owner)`
Should be:
`notify_ghosts(header = "Tactical Map", message = "The [faction] tactical map has been updated.", ghost_sound = "sound/effects/sos-morse-code.ogg", notify_volume = 80, action = NOTIFY_USCM_TACMAP, enter_link = "uscm_tacmap=1", enter_text = "View", source = owner)`

morrowwolf reviewed (changes requested):
Running list of bugs I am semi-aware of:
Crew manifest broken
Overwatch console not working
IFF not working properly (this should really be converted to the SG version, feel free to grab the component changes from main CM and port them elsewhere)
Motion detectors not working properly

AndroBetel commented:
> Crew manifest broken
 
fixed!
> Overwatch console not working

wasn't a problem
> IFF not working properly (this should really be converted to the SG version, feel free to grab the component changes from main CM and port them elsewhere)

iff fixed, fire prevention iff is made in another pr

> Motion detectors not working properly

fixed too

morrowwolf commented on code/datums/ammo/bullet/special_ammo.dm:
```diff
@@ -171,6 +171,6 @@
 	accurate_range = 14
-	damage = 35
+	damage = 45
 	penetration= ARMOR_PENETRATION_TIER_6
```
This pen should be lowered to 2

morrowwolf commented on code/modules/clothing/suits/marine_armor.dm:
```diff
@@ -1266,20 +1266,21 @@
 	armor_bio = CLOTHING_ARMOR_MEDIUMLOW
```
Make sure these are consistent with marine armor

AndroBetel replied:
upp armor is more resistant to bullets but their weapons hit stronger so i guess that's fine?

morrowwolf reviewed (commented):
Mapping/sprites seem fine

AndroBetel commented on code/game/machinery/vending/vendor_types/squad_prep/squad_prep.dm:
```diff
@@ -294,6 +377,74 @@
+		list("M89-S Signal Flare Pack", round(scale * 1), /obj/item/storage/box/m94/signal, VENDOR_ITEM_REGULAR),
+
+		list("SIDEARMS", -1, null, null),
+		list("NP92 Standard Pistol", round(scale * 2), /obj/item/weapon/gun/pistol/np92, VENDOR_ITEM_REGULAR),
```
they're pretty much the same as uscm's weapons but nobody takes them anyway because the upp revolver just sounds cool as fuck 😭

AndroBetel commented on code/modules/gear_presets/uscm.dm:
```diff
@@ -392,6 +463,25 @@
+	name = "UPP Platoon Sergeant"
+	assignment = JOB_SQUAD_LEADER_UPP
+	access = list(ACCESS_UPP_GENERAL, ACCESS_UPP_LEADERSHIP, ACCESS_UPP_FLIGHT)
+	languages = list(LANGUAGE_RUSSIAN, LANGUAGE_ENGLISH, LANGUAGE_CHINESE)
```
gonna change it
